compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```diff
@@ -1440,6 +1441,219 @@ LogicalResult ArgCompareOp::generateScalarImplementation(OpBuilder &b,
+  Operation *valueBroadcastOp = linalg::BroadcastOp::create(
+      b, loc, valueInit, emptyValueTensor, broadcastDims);
+  Operation *indexBroadcastOp = linalg::BroadcastOp::create(
+      b, loc, indexInit, emptyIndexTensor, broadcastDims);
+
+  return SmallVector<Value>{valueBroadcastOp->getResult(0),
+                            indexBroadcastOp->getResult(0)};
+}
+
+FailureOr<TilingResult> ArgCompareOp::tileToPartialReduction(
+    OpBuilder &b, Location loc, ReductionTilingStrategy strategy,
+    ValueRange init, ArrayRef<OpFoldResult> offsets,
+    ArrayRef<OpFoldResult> sizes,
+    const llvm::SetVector<unsigned> &reductionDims,
+    ArrayRef<OpFoldResult> splitReductionIvs) {
+  if (strategy != ReductionTilingStrategy::PartialReductionOuterParallel) {
+    return failure();
+  }
+  OpBuilder::InsertionGuard guard(b);
+
+  int64_t rank = getInputRank();
+  int64_t reductionDim = getDimension();
+
+  assert(offsets.size() == static_cast<size_t>(rank) &&
+         "Unexpected offsets size");
+  assert(sizes.size() == static_cast<size_t>(rank) && "Unexpected sizes size");
+
+  SmallVector<Operation *> slices;
+  SmallVector<Value> tiledOperands;
+
+  // Extract a slice of the input operand.
+  SmallVector<OpFoldResult> strides(rank, b.getIndexAttr(1));
+  Operation *inputSlice =
+      getSlice(b, loc, getInputValue(), offsets, sizes, strides);
+  tiledOperands.push_back(inputSlice->getResult(0));
+  slices.push_back(inputSlice);
+
+  // Extract slices of the init operands (partial results). For split-reduction,
+  // slice along the reduction dimension to get extra parallelism.
+  SmallVector<OpFoldResult> initOffsets, initSizes, initStrides;
+  for (int64_t i = 0; i < rank; ++i) {
+    if (i == reductionDim) {
+      initOffsets.push_back(splitReductionIvs[0]);
+      initSizes.push_back(b.getIndexAttr(1));
+    } else {
+      // For non-reduction dimensions, use the same offsets/sizes as input.
+      initOffsets.push_back(offsets[i]);
+      initSizes.push_back(sizes[i]);
+    }
+    initStrides.push_back(b.getIndexAttr(1));
+  }
+
+  SmallVector<int64_t> resultValShape, resultIdxShape;
+  for (int64_t i = 0; i < rank; ++i) {
+    if (i == reductionDim) {
+      continue;
+    }
+
+    if (auto sizeAttr = dyn_cast<Attribute>(initSizes[i])) {
+      int64_t size = (cast<IntegerAttr>(sizeAttr)).getInt();
+      resultValShape.push_back(size);
+      resultIdxShape.push_back(size);
+      continue;
+    }
+
+    resultValShape.push_back(ShapedType::kDynamic);
+    resultIdxShape.push_back(ShapedType::kDynamic);
+  }
+
+  auto sliceValResultType = RankedTensorType::get(
+      resultValShape, getOutputValueType().getElementType(),
+      cast<RankedTensorType>(getOutputValueType()).getEncoding());
+  auto sliceIdxResultType = RankedTensorType::get(
+      resultIdxShape, getOutputIndexType().getElementType(),
+      cast<RankedTensorType>(getOutputIndexType()).getEncoding());
```

Comment:
I'm not sure if we can handle encodings or not. They are tricky. I think we can ignore the encodings like other operations, so it becomes simpler.

```suggestion
  auto sliceValResultType = RankedTensorType::get(
      resultValShape, getOutputValueType().getElementType());
  auto sliceIdxResultType = RankedTensorType::get(
      resultIdxShape, getOutputIndexType().getElementType());
```

It looks like it can be simpler with using the [clone](https://github.com/llvm/llvm-project/blob/4eadb45f83cef00165055f8038f179ca5c3e88ef/mlir/include/mlir/IR/BuiltinTypeInterfaces.td#L246-L251) method. E.g.,

```cpp
ShapedType sliceValResultType = getOutputValueType().clone(resultValShape);
// ...
```